crates/kitsune_p2p/kitsune_p2p/src/spawn/actor.rs

```diff
@@ -284,6 +286,21 @@ impl KitsuneP2pActor {
         // TODO - use a real config
         let fetch_queue = FetchQueue::new_bitwise_or();
 
+        // Start a loop to handle our fetch queue fetch items.
+        {
+            let fetch_queue = fetch_queue.clone();
+            let i_s = internal_sender.clone();
+            tokio::task::spawn(async move {
+                tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+
+                for (key, space, source) in fetch_queue.get_items_to_fetch() {
+                    if let Err(err) = i_s.fetch(key, space, source).await {
```

Comment:
Would this be the place to enforce some kind of rate limiting? Like ensuring we never have more than X bytes of ops worth of fetch requests out at once??

Reply:
mmm, maybe that would make more sense inside the fetch queue? It can know what ops it returns in the iterator, and we wouldn't have to have external state / send back the sizes then?